Part2/Part2.ipynb

```diff
@@ -0,0 +1,1331 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, define the material properties and boundary conditions related to the thermal-hydraulic solver.\n",
+    "These values shown here roughly correspond to Pressurized Water Reactor (PWR) conditions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "R = 0.97 / 2.0           # outer radius of the pincell (cm)\n",
+    "Rf = 0.825 / 2.0         # outer radius of the pellet (cm)\n",
+    "pitch = 1.28             # pitch between pincells (cm)\n",
+    "height = 381.0           # height of the pincell (cm)\n",
+    "\n",
+    "htc = 0.5                # fluid-solid heat transfer coefficient (W/cm2/K)\n",
+    "mdot = 300.0             # mass flowrate (g/s)\n",
+    "power = 60.0e3           # total power produced by pincell (W)\n",
+    "T_inlet = 573.0          # inlet water temperature (K)\n",
+    "\n",
+    "UO2_k = 0.035            # UO2 thermal conductivity (W/cm/K)\n",
+    "water_cp = 4.184         # UO2 specific heat capacity (J/g/K)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, specify the number of layers we want to build in our model. Note that the more layers you add, the more particles you will need to see acceptable tally statistics in each layer!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "NameError",
+     "evalue": "name 'height' is not defined",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
+      "\u001b[0;32m<ipython-input-4-7317ff0522cd>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mN\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m10\u001b[0m                    \u001b[0;31m# number of coupling layers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mH\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mheight\u001b[0m \u001b[0;34m/\u001b[0m \u001b[0mN\u001b[0m            \u001b[0;31m# height of each coupling layer\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
+      "\u001b[0;31mNameError\u001b[0m: name 'height' is not defined"
+     ]
+    }
+   ],
+   "source": [
+    "N = 10                    # number of coupling layers         \n",
+    "H = height / N            # height of each coupling layer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following functions comprise our T/H app:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def water_density(T):\n",
```

Comment:
The reason I chose the function I did is that there's no errors if you get above saturation or anything (obviuosly the results are wrong), but I didn't want that to derail people if they changed the inlet BCs, etc. if the purpose of the tutorial is just to demonstrate the C-API

Reply:
I think I'll keep the geometry and material parts in the notebook though, since in order to change densities we need to make N separate water materials